src/Adapter/Shop/MaintenanceConfiguration.php

```diff
@@ -61,24 +51,40 @@ public function getConfiguration()
      */
     public function updateConfiguration(array $configuration)
     {
-        if ($this->validateConfiguration($configuration)) {
-            $this->configuration->set('PS_SHOP_ENABLE', $configuration['enable_shop']);
-            $this->configuration->set('PS_MAINTENANCE_IP', $configuration['maintenance_ip']);
-            $this->configuration->set('PS_MAINTENANCE_TEXT', $configuration['maintenance_text'], null, ['html' => true]);
+        $shopConstraint = null;
+        if (!$this->shopContext->isAllShopContext()) {
+            $configuration = $this->removeDisabledFields($configuration);
+            $contextShopGroup = $this->shopContext->getContextShopGroup();
+            $contextShopId = $this->shopContext->getContextShopID();
+            $contextShopId = (int) $contextShopId > 0 ? $contextShopId : null;
+
+            $shopConstraint = new ShopConstraint(
+                $contextShopId,
+                $contextShopGroup->id
+            );
+        }
+
+        if (isset($configuration['enable_shop'])) {
```

Comment:
Likewise this code could be in a helper function from the abstract class:

`protected function updateConfigurationValue(string $configurationKey, string $formKey, array $input, ?ShopConstraint $shopConstraint): void`